crates/hir/src/lib.rs

```diff
@@ -1074,6 +1083,10 @@ impl Trait {
         Module { id: self.id.lookup(db.upcast()).container }
     }
 
+    pub fn krate(self, db: &dyn HirDatabase) -> Crate {
+        self.module(db).krate()
+    }
```

Comment:
It's better to  avoid adding these in the hir, and let the caller call `module.krate()` themselves.

Reply:
Some of the `krate` methods were there before; should I remove them and replace the callers with `.module(db).krate()` too?